packages/zend-locale/library/Zend/Locale/Format.php

```diff
@@ -929,7 +929,7 @@ private static function _parseDate($date, $options)
                         }
                     } else {
                         // iconv_substr changes since 7.0.11
-                        if (iconv_strlen($splitted[0][0]) === $split) {
+                        if (iconv_strlen($splitted[0][0]) <= $split) {
```

Comment:
note that `===` is strict compare but `<=` is not strict compare.

so the semantics will change if both sides are not of same type. have you considered this?

Reply:
$splitted is `preg_match_all('/\d+/u', $number, $splitted);` so we needn't worry about about its type. $split is either false or int and the false case is always checked explicitly.